tidy3d/plugins/microwave/path_integrals.py

```diff
@@ -18,16 +19,34 @@
     TimeDataArray,
 )
 from ...components.data.monitor_data import FieldData, FieldTimeData, ModeSolverData
-from ...components.geometry.base import Box
-from ...components.types import Axis, Direction
+from ...components.geometry.base import Box, Geometry
+from ...components.types import Ax, Axis, Coordinate2D, Direction
 from ...components.validators import assert_line, assert_plane
+from ...components.viz import add_ax_if_none
+from ...constants import AMP, VOLT, fp_eps
 from ...exceptions import DataError, Tidy3dError
+from .viz import (
+    ARROW_CURRENT,
+    plot_params_current_path,
+    plot_params_voltage_minus,
+    plot_params_voltage_path,
+    plot_params_voltage_plus,
+)
 
 MonitorDataTypes = Union[FieldData, FieldTimeData, ModeSolverData]
 EMScalarFieldType = Union[ScalarFieldDataArray, ScalarFieldTimeDataArray, ScalarModeFieldDataArray]
 IntegralResultTypes = Union[FreqDataArray, FreqModeDataArray, TimeDataArray]
 
 
+def _check_em_field_supported(em_field: Any):
```

Comment:
@weiliangjin2021 @tylerflex I wonder if you have any ideas for checking class method parameters. I was experimenting with decorators and looked into pydantic support. decorators mess up the docs and pydantic support does not work in our version, I think in V2 it will be supported.

Reply:
as far as I understand, we use decorators for this kind of thing in a few places, but agree it's not great for the docs.
https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/medium.py#L95